src/Workers/Indexers/Inscriptions.ts

```diff
@@ -42,14 +42,21 @@ export const inscriptionsIndexer: IndexHandler = {
 
 async function getInscriptions(vins: VinData[]) {
   const envelopes: Envelope[] = [];
+  let currentTxid = "";
+  let currentEnvelopeIndex = 0;
   for (const vin of vins) {
-    const _envelopes = Envelope.fromTxinWitness(vin.txid, vin.witness);
+    if (vin.txid !== currentTxid) {
+      currentTxid = vin.txid;
+      currentEnvelopeIndex = 0;
+    }
```

Comment:
are the `getInscriptions` input, the Vins of a whole block? and not just a txs? If so, then could there be a chance that they are not ordered by `vin.txid`? (e.g [txid1-vin1, txid2-vin1, txid1-vin2]

if so then, I think might need a more robust solution 🤔 to keep track of the `currentEnvelopeIndex`

Reply:
Yup, our Indexer extracts the Vins from a whole block when we are indexing. Under the current implementation, it is correctly ordered for this implementation to work. (eg. [txid1-vin1, txid1-vin2, txid2-vin1])

I had the same concern so I took a few days to think about it, but I think this works for now. The alternative is to use a map, but that will probably slow down the indexer.

Reply:
I am actually more worried about the general case where the ordering is not preserved (eg. [txid1-vin2, txid2-vin1, txid1-vin1]), but I can't think of a good solution to this right now without an extensive rewrite of the code.

Reply:
Since this is strictly ordered, I'll merge this in to test first. We'll update this if we rework the Indexer logic.